tests/suite/test6_arg_kwarg_combinations.py

```diff
@@ -0,0 +1,342 @@
+		
+		pub.sendMessage('echo', arg1 = 1, kwarg1 = 3, kwarg2 = 5)
+		self.assertEqual(self.getReply(), (1, 2, 3, 5))
+		
+		pub.sendMessage('echo', arg1 = 1, arg2 = 5, kwarg1 = 3)
+		self.assertEqual(self.getReply(), (1, 5, 3, 4))
+		
+		pub.sendMessage('echo', arg1 = 1, arg2 = 5, kwarg1 = 3, kwarg2 = 6)
+		self.assertEqual(self.getReply(), (1, 5, 3, 6))
+
+	def test_args_withDefaults_kwargs_withDefaults_starArgs(self):
+		def myFunction(arg1, arg2 = 2, *args, kwarg1, kwarg2 = 4): self.answer = (arg1, arg2, args, kwarg1, kwarg2)
+		pub.subscribe(myFunction, 'echo')
+		
+		pub.sendMessage('echo', arg1 = 1, kwarg1 = 3)
+		self.assertEqual(self.getReply(), (1, 2, (), 3, 4))
+		
+		pub.sendMessage('echo', arg1 = 1, kwarg1 = 3, kwarg2 = 5)
+		self.assertEqual(self.getReply(), (1, 2, (), 3, 5))
+		
+		pub.sendMessage('echo', arg1 = 1, arg2 = 5, kwarg1 = 3)
+		self.assertEqual(self.getReply(), (1, 5, (), 3, 4))
+		
+		pub.sendMessage('echo', arg1 = 1, arg2 = 5, kwarg1 = 3, kwarg2 = 6)
+		self.assertEqual(self.getReply(), (1, 5, (), 3, 6))
+
+	def test_args_withDefaults_kwargs_withDefaults_starStarKwargs(self):
+		def myFunction(arg1, arg2 = 2, *, kwarg1, kwarg2 = 4, **kwargs): self.answer = (arg1, arg2, kwarg1, kwarg2, kwargs)
+		pub.subscribe(myFunction, 'echo')
+		
+		pub.sendMessage('echo', arg1 = 1, kwarg1 = 3)
+		self.assertEqual(self.getReply(), (1, 2, 3, 4, {}))
+		
+		pub.sendMessage('echo', arg1 = 1, kwarg1 = 3, kwarg2 = 5)
+		self.assertEqual(self.getReply(), (1, 2, 3, 5, {}))
+		
+		pub.sendMessage('echo', arg1 = 1, arg2 = 5, kwarg1 = 3)
+		self.assertEqual(self.getReply(), (1, 5, 3, 4, {}))
+		
+		pub.sendMessage('echo', arg1 = 1, arg2 = 5, kwarg1 = 3, kwarg2 = 6)
+		self.assertEqual(self.getReply(), (1, 5, 3, 6, {}))
+
+	def test_args_withDefaults_kwargs_withDefaults_starArgs_starStarKwargs(self):
+		def myFunction(arg1, arg2 = 2, *args, kwarg1, kwarg2 = 4, **kwargs): self.answer = (arg1, arg2, args, kwarg1, kwarg2, kwargs)
+		pub.subscribe(myFunction, 'echo')
+		
+		pub.sendMessage('echo', arg1 = 1, kwarg1 = 3)
+		self.assertEqual(self.getReply(), (1, 2, (), 3, 4, {}))
+		
+		pub.sendMessage('echo', arg1 = 1, kwarg1 = 3, kwarg2 = 5)
+		self.assertEqual(self.getReply(), (1, 2, (), 3, 5, {}))
+		
+		pub.sendMessage('echo', arg1 = 1, arg2 = 5, kwarg1 = 3)
+		self.assertEqual(self.getReply(), (1, 5, (), 3, 4, {}))
+		
+		pub.sendMessage('echo', arg1 = 1, arg2 = 5, kwarg1 = 3, kwarg2 = 6)
+		self.assertEqual(self.getReply(), (1, 5, (), 3, 6, {}))
+
+	def test_classMethod(self):
+		class myClass():
+			def __init__(self, parent):
+				self.parent = parent
+			def myFunction(self, arg1, arg2 = 2): self.parent.answer = (self, arg1, arg2)
+		myObject = myClass(self)
+		pub.subscribe(myObject.myFunction, 'echo')
+		
+		pub.sendMessage('echo', arg1 = 1, arg2 = 3)
+		reply = self.getReply()
+
+		self.assertEqual(reply[1:], (1, 3))
+		self.assertIsInstance(reply[0], myClass)
+					
+if __name__ == '__main__':
```

Comment:
Once the test is converted to be run by pytest, the "main" block will be obsolete.